health/health.go

```diff
@@ -0,0 +1,45 @@
+package health
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/featureform/metadata"
+	"github.com/featureform/provider"
+	pc "github.com/featureform/provider/provider_config"
+	pt "github.com/featureform/provider/provider_type"
+)
+
+type Health struct {
+	metadata *metadata.Client
+}
+
+func NewHealth(client *metadata.Client) *Health {
+	return &Health{
+		metadata: client,
+	}
+}
+
+func (h *Health) Check(providerName string) (bool, error) {
```

Comment:
nit: what do you think about CheckProviderHealth

Reply:
who's going to call this? We have to expose protos for this right?

Reply:
Implementing this as a service with its own API is out of scope for this initial phase, but in the near future, we'll need protos for this, yes.